pandas/core/ops.py

```diff
@@ -276,15 +276,35 @@ def __init__(self, left, right, name):
 
         self.left = left
         self.right = right
-        lvalues = self._convert_to_array(left, name=name)
-        rvalues = self._convert_to_array(right, name=name, other=lvalues)
 
-        self.is_timedelta_lhs = is_timedelta64_dtype(left)
-        self.is_datetime_lhs = is_datetime64_dtype(left)
-        self.is_integer_lhs = left.dtype.kind in ['i', 'u']
-        self.is_datetime_rhs = is_datetime64_dtype(rvalues)
-        self.is_timedelta_rhs = is_timedelta64_dtype(rvalues)
-        self.is_integer_rhs = rvalues.dtype.kind in ('i', 'u')
+
+        if self._is_offset(left):
+            lvalues = left
+            self.is_offset_lhs = True
```

Comment:
maybe some of these up top to make this less repetive, e.g. (e.g. the 3 ``False`` ones)

```
self.is_datetime_lhs = False
....
```